shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeIngressController.java

```diff
@@ -85,61 +100,116 @@ public List<EventSource> prepareEventSources(EventSourceContext<BridgeIngress> e
+            status.markConditionTrue(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE);
         }
 
-        bridgeIngressService.fetchOrCreateBridgeIngressAuthorizationPolicy(bridgeIngress, path);
-
         // Nothing to check for Authorization Policy
+        bridgeIngressService.fetchOrCreateBridgeIngressAuthorizationPolicy(bridgeIngress, path);
+        if (!status.isConditionTypeTrue(BridgeIngressStatus.AUTHORISATION_POLICY_AVAILABLE)) {
+            status.markConditionTrue(BridgeIngressStatus.AUTHORISATION_POLICY_AVAILABLE);
+        }
 
         NetworkResource networkResource = networkingService.fetchOrCreateBrokerNetworkIngress(bridgeIngress, secret, path);
 
         if (!networkResource.isReady()) {
             LOGGER.info("Ingress networking resource BridgeIngress: '{}' in namespace '{}' is NOT ready", bridgeIngress.getMetadata().getName(),
                     bridgeIngress.getMetadata().getNamespace());
-            bridgeIngress.getStatus().markConditionFalse(ConditionTypeConstants.READY);
-            bridgeIngress.getStatus().markConditionTrue(ConditionTypeConstants.AUGMENTATION, ConditionReasonConstants.NETWORK_RESOURCE_NOT_READY);
-            return UpdateControl.updateStatus(bridgeIngress);
+            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                status.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            if (!status.isConditionTypeFalse(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE)) {
+                status.markConditionFalse(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE);
+            }
+            return UpdateControl.updateStatus(bridgeIngress).rescheduleAfter(ingressPollIntervalSeconds);
+        } else if (!status.isConditionTypeTrue(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE)) {
+            status.markConditionTrue(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE);
         }
 
         LOGGER.info("Ingress networking resource BridgeIngress: '{}' in namespace '{}' is ready", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
 
         // Only issue a Status Update once.
         // This is a work-around for non-deterministic Unit Tests.
         // See https://issues.redhat.com/browse/MGDOBR-1002
-        if (!bridgeIngress.getStatus().isReady()) {
-            bridgeIngress.getStatus().markConditionTrue(ConditionTypeConstants.READY);
-            bridgeIngress.getStatus().markConditionFalse(ConditionTypeConstants.AUGMENTATION);
+        if (!status.isReady()) {
+            status.markConditionTrue(ConditionTypeConstants.READY);
             notifyManager(bridgeIngress, ManagedResourceStatus.READY);
             return UpdateControl.updateStatus(bridgeIngress);
         }
 
         return UpdateControl.noUpdate();
     }
 
+    private boolean isTimedOut(BridgeIngressStatus status) {
+        Optional<Date> lastTransitionDate = status.getConditions()
```

Comment:
Each `Condition` will have it's own "last transitioned date". I therefore get the _newest_ value when considering if provisioning has timed out as a whole.